tests/e2e/backup_restore_suite_test.go

```diff
@@ -125,34 +137,62 @@ func runBackupAndRestore(brCase BackupRestoreCase, expectedErr error, updateLast
+		pvcPath = fmt.Sprintf(pvcPathFormat, brCase.Namespace, pvcName)
 
 		err = InstallApplication(dpaCR.Client, pvcPath)
 		Expect(err).ToNot(HaveOccurred())
 	}
 
 	// wait for pods to be running
-	Eventually(AreAppBuildsReady(dpaCR.Client, brCase.ApplicationNamespace), timeoutMultiplier*time.Minute*5, time.Second*5).Should(BeTrue())
-	Eventually(AreApplicationPodsRunning(kubernetesClientForSuiteRun, brCase.ApplicationNamespace), timeoutMultiplier*time.Minute*9, time.Second*5).Should(BeTrue())
+	Eventually(AreAppBuildsReady(dpaCR.Client, brCase.Namespace), timeoutMultiplier*time.Minute*5, time.Second*5).Should(BeTrue())
+	Eventually(AreApplicationPodsRunning(kubernetesClientForSuiteRun, brCase.Namespace), timeoutMultiplier*time.Minute*9, time.Second*5).Should(BeTrue())
 
-	// Run optional custom verification
-	log.Printf("Running pre-backup function for case %s", brCase.Name)
-	err = brCase.PreBackupVerify(dpaCR.Client, brCase.ApplicationNamespace)
+	// do the backup for real
+	nsRequiredResticDCWorkaround := runBackup(brCase.BackupRestoreCase, backupName)
+
+	// uninstall app
+	log.Printf("Uninstalling app for case %s", brCase.Name)
+	err = UninstallApplication(dpaCR.Client, brCase.ApplicationTemplate)
 	Expect(err).ToNot(HaveOccurred())
 
-	nsRequiresResticDCWorkaround, err := NamespaceRequiresResticDCWorkaround(dpaCR.Client, brCase.ApplicationNamespace)
+	// Wait for namespace to be deleted
+	Eventually(IsNamespaceDeleted(kubernetesClientForSuiteRun, brCase.Namespace), timeoutMultiplier*time.Minute*4, time.Second*5).Should(BeTrue())
+
+	updateLastInstallTime()
+
+	// run restore
+	runRestore(brCase.BackupRestoreCase, backupName, restoreName, nsRequiredResticDCWorkaround)
+
+	// verify app is running
+	Eventually(AreAppBuildsReady(dpaCR.Client, brCase.Namespace), timeoutMultiplier*time.Minute*3, time.Second*5).Should(BeTrue())
+	Eventually(AreApplicationPodsRunning(kubernetesClientForSuiteRun, brCase.Namespace), timeoutMultiplier*time.Minute*9, time.Second*5).Should(BeTrue())
+
+	// Run optional custom verification
+	if brCase.PostRestoreVerify != nil {
+		log.Printf("Running post-restore function for case %s", brCase.Name)
+		err = brCase.PostRestoreVerify(dpaCR.Client, brCase.Namespace)
+		Expect(err).ToNot(HaveOccurred())
+	}
+}
+
+func runBackup(brCase BackupRestoreCase, backupName string) bool {
+	// Run optional custom verification
+	if brCase.PreBackupVerify != nil {
+		log.Printf("Running pre-backup function for case %s", brCase.Name)
+		err := brCase.PreBackupVerify(dpaCR.Client, brCase.Namespace)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	nsRequiresResticDCWorkaround, err := NamespaceRequiresResticDCWorkaround(dpaCR.Client, brCase.Namespace)
 	Expect(err).ToNot(HaveOccurred())
 
-	// TODO this should be a function, not an arbitrary sleep
-	log.Printf("Sleeping for %v to allow application to be ready for case %s", brCase.AppReadyDelay, brCase.Name)
-	time.Sleep(brCase.AppReadyDelay)
```

Comment:
I think tests are not passing because of this https://github.com/openshift/oadp-operator/commit/355cf0f5425dceb10ded6f7954daf561f72c74f7#diff-a9f567f9b6083a8f8e52d9318806042225c729fa99bea2354e148eead33bdf9fR189-R195

Reply:
Thanks! I am going to try pulling in the replacement from: https://github.com/openshift/oadp-operator/commit/355cf0f5425dceb10ded6f7954daf561f72c74f7#diff-a9f567f9b6083a8f8e52d9318806042225c729fa99bea2354e148eead33bdf9fL192